scipy/optimize/_zeros_py.py

```diff
@@ -1392,3 +1393,344 @@ def toms748(f, a, b, args=(), k=1,
+    if not np.issubdtype(np.result_type(x1, x2, f1, f2), np.floating):
+        raise ValueError("Bracket and function output must be real numbers.")
+    x1, f1, x2, f2 = x1.astype(ft), f1.astype(ft), x2.astype(ft), f2.astype(ft)
+
+    # To ensure that we can do indexing, we'll work with at least 1d arrays,
+    # but remember the appropriate shape of the output.
+    shape = x1.shape
+    x1, f1, x2, f2, = x1.ravel(), f1.ravel(), x2.ravel(), f2.ravel()
+    return x1, f1, x2, f2, shape, ft
+
+
+def _check_termination(x1, f1, x2, f2, x3, f3,
+                       res, active, status, nfev, nit, tols):
+    # Check for all terminal conditions and record statuses.
+
+    # See [1] Section 4
+    i = np.abs(f1) < np.abs(f2)
+    xmin = np.choose(i, (x2, x1))
+    fmin = np.choose(i, (f2, f1))
+    stop = np.zeros_like(x1, dtype=bool)  # termination condition met
+
+    # This is the convergence criterion used in bisect. Chandrupatla's
+    # criterion is equivalent to abs(x2-x1) < 4*abs(xmin)*xrtol + xatol.
+    dx = abs(x2 - x1)
+    tol = abs(xmin)*tols['xrtol'] + tols['xatol']
+    i = dx < tol
+    # Tolerance on function value. Note that `frtol` has been redefined as
+    # `frtol = frtol * np.minimum(f1, f2)`, where `f1` and `f2` are the
+    # function evaluated at the original ends of the bracket.
+    i |= np.abs(fmin) <= tols['fatol'] + tols['frtol']
+    stop[i], status[i] = True, _ECONVERGED
+
+    i = (np.sign(f1) == np.sign(f2)) & ~stop
+    xmin[i], fmin[i], stop[i], status[i] = np.nan, np.nan, True, _ESIGNERR
+
+    i = ~((np.isfinite(x1) & np.isfinite(x2)
+           & np.isfinite(f1) & np.isfinite(f2)) | stop)
+    xmin[i], fmin[i], stop[i], status[i] = np.nan, np.nan, True, _EVALUEERR
+
+    if np.any(stop):
+        # update the result object with the elements for which termination
+        # condition has been met
+        active_stop = active[stop]
+        res.x[active_stop] = xmin[stop]
+        res.fun[active_stop] = fmin[stop]
+        res.xl[active_stop] = x1[stop]
+        res.xr[active_stop] = x2[stop]
+        res.fl[active_stop] = f1[stop]
+        res.fr[active_stop] = f2[stop]
+        res.status[active_stop] = status[stop]
+        res.nfev[active_stop] = nfev
+        res.nit[active_stop] = nit
+        res.success[active_stop] = res.status[active_stop] == 0
+
+        # compress the arrays to avoid unnecessary computation
+        proceed = ~stop
+        active = active[proceed]
+        xmin = xmin[proceed]
+        fmin = fmin[proceed]
+        x1 = x1[proceed]
+        f1 = f1[proceed]
+        x2 = x2[proceed]
+        f2 = f2[proceed]
+        x3 = x3[proceed] if x3 is not None else x3
+        f3 = f3[proceed] if f3 is not None else f3
+        status = status[proceed]
+        tols['frtol'] = tols['frtol'][proceed]
+        tol = tol[proceed]
+        dx = dx[proceed]
+
+    # See [1] Appendix, "Adjust T Away from the Interval Boundary"
+    tl = 0.5 * tol / dx
```

Comment:
Because it depends on `tol` and `dx`, which are both computed in here naturally as a part of the convergence check. if we don't calculate `tl` in here, we need to return both `tol` and `dx`, which would only be used to calculate `tl`.